### PR TITLE
Fixes empty hostname returned from HostDiscoveryScript

### PR DESCRIPTION
**Pull request by plliao**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes empty hostname returned from HostDiscoveryScript and make sure hostname is not empty.

Context:
When discovery script return nothing, HostDiscoveryScript will treat the empty result i.e. "" string as the available host.
It will fail training job as launcher cannot ssh on the host with empty address. Discovery script can return nothing when no workers are available or new workers are not ready yet.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: test/single/test_elastic_discovery.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    def __init__(self, *args, **kwargs):
+        super(HostDiscoveryScriptTests, self).__init__(*args, **kwargs)
+
+    @parameterized.expand([
+        ["", {}],  # No host found.
+        ["host-1", {"host-1": DEFAULT_SLOTS}],
+        ["host-1:2", {"host-1": 2}],
+        ["host-1\nhost-5", {"host-1": DEFAULT_SLOTS, "host-5": DEFAULT_SLOTS}],
+        ["host-1:2\nhost-4:3", {"host-1": 2, "host-4": 3}],
```
Comment (EnricoMi) [inline test/single/test_elastic_discovery.py:39]: Can we also add tests where multiple hosts exist, but also an empty line, because the new code should handle that case gracefully as well:

    ["\nhost-1:2\nhost-4:3", ...],
    ["host-1:2\n\nhost-4:3", ...],
    ["host-1:2\nhost-4:3\n", ...],

**Review by EnricoMi — CHANGES_REQUESTED** [review]

File: test/single/test_elastic_discovery.py (hunk 1 above)
Comment (plliao) [reply test/single/test_elastic_discovery.py:39]: Sure! I added those cases. Please review again. Thanks!

**Review by EnricoMi — APPROVED** [review]
LGTM!
